src/V3EmitC.cpp

```diff
@@ -1225,6 +1225,34 @@ class EmitCStmts VL_NOT_FINAL : public EmitCBaseVisitor {
             emitConstant(nodep, nullptr, "");
         }
     }
+    virtual void visit(AstStdRandomize* nodep) override {
+        auto* dtypep = nodep->varMemberp() ? nodep->varMemberp()->subDTypep()->subDTypep()
+                                           : nodep->varRefp()->dtypep()->subDTypep();
+        if (auto* enumDTypep = nodep->varEnumDTypep()) {
+            size_t itemCount = 0;  // Number of enum items
+            for (auto* itemp = enumDTypep->itemsp(); itemp;
+                 itemp = VN_CAST(itemp->nextp(), EnumItem))
+                itemCount++;
+            puts("VL_RANDOMIZE_ENUM_VAR(");
+            iterateAndNextNull(nodep->varValueTabRefp());
+            puts(", ");
+            puts(cvtToStr(itemCount));
+            puts(", ");
+        } else {
+            puts("VL_RANDOMIZE_VAR(");
+        }
+        iterateAndNextNull(nodep->varRefp());
+        if (nodep->varMemberp()) {
+            puts(", ");
+            puts(cvtToStr(nodep->varMemberLSB()));
+            puts(", ");
+            puts(cvtToStr(nodep->varMemberp()->width()));
+        } else if (!nodep->varEnumDTypep()) {
+            puts(", ");
+            puts(cvtToStr(nodep->varRefp()->width()));
+        }
+        puts(")");
+    }
```

Comment:
I think this can all go away and perhaps even AstStdRandomize if it instead builds lower level calls.

Also this shouldn't be building the enum table, that should be a single static table built similar to how V3Width did it.  It looks like you added code for that, so not sure why this was even needed.

Reply:
It doesn't build the enum table, it only counts the number of enum items. It uses the pre-existing table.

Not sure what you mean by "this can all go away".

Reply:
I mean if rather than RandomizeVisitor constructing AstStdRandomize, it would constuct AstAssign(AstRand, AstVarRef) or whatever more complicated for Enums. Then there's no AstStdRandomize's to emit any more, and no VL_RANDOMIZE_VAR in verilated.h.  Will comment elsewhere on meta-idea.